app/src/main/java/com/example/pinterbest/LoginFragment.kt

```diff
@@ -24,30 +36,127 @@ class LoginFragment : Fragment() {
+                    )
+                ).get(LoginViewModel::class.java)
+
+                model.logInCodeLiveData.observe(viewLifecycleOwner) { response ->
+                    loginProvider(view, response)
+                }
             } else {
-                Toast.makeText(context, "Not correct", Toast.LENGTH_SHORT).show()
-                it.findNavController().navigate(R.id.action_loginFragment_to_homeFragment)
+                Toast.makeText(context, "Произошла ошибка сети!", Toast.LENGTH_SHORT).show()
             }
         }
+
+        binding.emailAuth.setOnClickListener {
+            it.findNavController().navigate(R.id.action_loginFragment_to_registrationFragment)
+        }
     }
 
     override fun onDestroyView() {
         super.onDestroyView()
         _binding = null
     }
 
-    private fun isValidLogIn(): Boolean {
-        return Validator.isValidName(binding.usernameBox.text.toString().trim()) &&
-            Validator.isValidPassword(binding.passwordBox.text.toString().trim())
+    private fun setUpBottomNavigationItem() {
+        (activity as MainActivity).binding.bottomNavigation.menu
+            .getItem(MainActivity.HOME_POSITION_BNV).isChecked = true
+    }
+
+    private fun loginProvider(view: View, response: Response<ResponseBody>) {
+        when (response.code()) {
+            SUCCESS -> {
+                saveSession(response)
+                view.findNavController()
+                    .navigate(R.id.action_loginFragment_to_homeFragment)
+                setUpBottomNavigationItem()
+            }
+            INVALID_DATA ->
+                {
+                    Toast.makeText(
+                        context,
+                        "Предоставлены неверные учетные данные",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+            WRONG_PASSWORD ->
+                {
+                    Toast.makeText(
+                        context,
+                        "Неправильный пароль",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+            ALREADY_AUTHORIZED ->
+                {
+                    Toast.makeText(
+                        context,
+                        "Вы уже авторизованы. Сначала выйдите из системы",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+            USER_NOT_FOUND ->
+                {
+                    Toast.makeText(
+                        context,
+                        "Пользователь не найден",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+        }
+    }
+
+    private fun saveSession(response: Response<ResponseBody>) {
+        val regex =
+            """['session_id=']+(?<=session_id=).{56}""".toRegex()
+        val cookie = regex.find(response.headers().values("Set-Cookie")[0])
+        sessionManager.saveUserData(cookie?.value ?: "")
```

Comment:
UI layer is bad place for this logic. You need interactor for it. Also expressions with regex look weird, why don't you just access `Set-Cookie` header?
In case there is no way to implement it without this I suggest to consider some CookieProvider with the following contract

```kotlin
interface CookieProvider {
    // returns cookie string
    fun getCookie(response: Response<*>): String
}
```

Reply:
Created AuthRepository